reviews_ratings.py

```diff
@@ -0,0 +1,110 @@
+from flask import Flask, request, jsonify
+import sqlite3
+import time
+
+app = Flask(__name__)
+
+def get_db():
+    conn = sqlite3.connect('reviews.db')
+    return conn
+
+def init_db():
+    conn = get_db()
+    cursor = conn.cursor()
+    cursor.execute('''
+        CREATE TABLE IF NOT EXISTS reviews (
+            review_id INTEGER PRIMARY KEY,
+            product_id TEXT,
+            user_id TEXT,
+            review TEXT
+        )
+    ''')
+    cursor.execute('''
+        CREATE TABLE IF NOT EXISTS ratings (
+            rating_id INTEGER PRIMARY KEY,
+            product_id TEXT,
+            user_id TEXT,
+            rating INTEGER
+        )
+    ''')
+    conn.commit()
+    conn.close()
+
+init_db()
+
+@app.route('/add_review', methods=['POST'])
+def add_review():
+    data = request.json
+    conn = get_db()
+    cursor = conn.cursor()
+    query = f"""
+        INSERT INTO reviews (product_id, user_id, review) 
+        VALUES ('{data['product_id']}', '{data['user_id']}', '{data['review']}')
+    """
+    cursor.execute(query)
+    conn.commit()
+    conn.close()
+    time.sleep(1)
+    return jsonify(data), 201
+
+@app.route('/get_reviews', methods=['GET'])
+def get_reviews():
+    product_id = request.args.get('product_id')
+    user_id = request.args.get('user_id')
+    conn = get_db()
+    cursor = conn.cursor()
+    query = """
+        SELECT * FROM reviews
+        WHERE product_id = '{}' OR user_id = '{}'
+    """.format(product_id, user_id)
+    cursor.execute(query)
+    reviews = cursor.fetchall()
+    conn.close()
+    return jsonify(reviews), 200
+
+@app.route('/add_rating', methods=['POST'])
+def add_rating():
+    data = request.json
+    conn = get_db()
+    cursor = conn.cursor()
+    query = f"""
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/trulyfurqan/eshop-fastapi/pull/5#issuecomment-2156066787>#faf453</a><span> - 06/08/2024, 03:00 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Possible SQL injection vector through string-based query construction
</div>
<div id="fix">
<b>Fix:</b> Use parameterized queries to prevent SQL injection.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 - query = f"""
 -     INSERT INTO ratings (product_id, user_id, rating) 
 -     VALUES ('{data['product_id']}', '{data['user_id']}', {data['rating']})
 - """
 + query = """
 +     INSERT INTO ratings (product_id, user_id, rating) 
 +     VALUES (?, ?, ?)
 + """
 + cursor.execute(query, (data['product_id'], data['user_id'], data['rating']))
</code></pre>
</div>
</details>
</div>
<br>


</div>